src/content/app/tools/blast/components/listed-blast-submission/ListedBlastSubmission.tsx

```diff
@@ -143,30 +146,52 @@ const SequenceBox = (props: SequenceBoxProps) => {
       <div>Sequence {sequence.id}</div>
       <div className={styles.sequenceHeader}>{sequenceHeaderLabel}</div>
       <div className={styles.speciesCount}>
-        <span className={styles.againstText}>Against</span>
-        {speciesCount} species
+        <span className={styles.againstText}>Against</span> {speciesCount}{' '}
+        species
       </div>
-      <StatusElement jobs={jobs} />
+      {jobs.length ? (
+        <StatusElement sequenceId={sequence.id} jobs={jobs} />
+      ) : null}
     </div>
   );
 };
 
-const StatusElement = ({ jobs }: { jobs: BlastJob[] }) => {
-  const hasRunningJobs = jobs.some((job) => job.status === 'RUNNING');
-  const hasFailedJobs = jobs.some((job) => job.status === 'FAILURE');
+type StatusElementProps = {
+  sequenceId?: number;
+  jobs: BlastJob[];
+};
+
+export const StatusElement = (props: StatusElementProps) => {
+  const { sequenceId, jobs } = props;
+
+  isFailedBlastSubmission();
+
+  const hasRunningJobs = jobs.some(
+    (job) =>
+      job.status === 'RUNNING' &&
+      (sequenceId ? sequenceId === job.sequenceId : true)
+  );
+  const hasAllFailedJobs = jobs.every(
```

Comment:
maybe `hasAllFailedJobs` -> `hasOnlyFailedJobs`? Although I am not feeling strongly about this one.

Reply:
or `hasAllJobsFailed` ? :-)

Reply:
> or hasAllJobsFailed ?

I don't know what that means — does it mean that (1) something (e.g. a box) only **has** jobs that failed, or (2) that all jobs **have** failed? From the word order, it sounds like the second; but then the verb needs to be different.